### PR TITLE
sysconfig/cloudinit.go: measure (but don't use) gadget cloud-init datasource

### PR DESCRIPTION
**Pull request by anonymouse64**
In a future commit, we will use the result of measuring the datasource settings
for the cloud-init config from the gadget, but for now just introduce the
machinery to read it.

### CODE REVIEW

**Comment by anonymouse64** [comment]
I opened https://github.com/snapcore/snapd/pull/10573 which is the full WIP branch that is based on top of this to demonstrate how this code is used, specifically see https://github.com/snapcore/snapd/pull/10573/files#diff-3450862368a77e8ae5d3817feb24a674eadc445a262c1f086d5a3e1a6174ad21L185-R446 for how this is used for decision making

File: sysconfig/cloudinit_test.go [hunk 1]
```diff
@@ -655,3 +655,126 @@ func (s *sysconfigSuite) TestRestrictCloudInit(c *C) {
...
+
+func (s *sysconfigSuite) TestCloudDatasourcesInUse(c *C) {
+	tt := []struct {
+		configFileContent string
+		expError          string
+		expRes            *sysconfig.CloudDatasourcesInUseResult
+		comment           string
+	}{
+		{
+			configFileContent: `datasource_list: [MAAS]`,
```
Comment (pedronis) [inline sysconfig/cloudinit_test.go:701]: do we need a test where we have more than one value in the list?

File: sysconfig/cloudinit.go [hunk 2]
```diff
@@ -63,6 +67,105 @@ func DisableCloudInit(rootDir string) error {
...
+	}
+
+	// we can also have datasources mentioned in the datasource list config
+	if cfg.DatasourceList != nil {
+		for _, ds := range *cfg.DatasourceList {
+			sourcesMentionedInCfg[strings.ToUpper(ds)] = true
+		}
+
+		for _, ds := range *cfg.DatasourceList {
+			res.ExplicitlyAllowed = append(res.ExplicitlyAllowed, strings.ToUpper(ds))
```
Comment (pedronis) [inline sysconfig/cloudinit.go:152]: I suppose is not a problem if we allow for repetitions?

**Review by pedronis — APPROVED** [review]
lgtm

File: sysconfig/cloudinit.go (hunk 2 above)
Comment (anonymouse64) [reply sysconfig/cloudinit.go:152]: hmm it's probably not but I think we should handle it more consistently anyways, I will fix this and add a test for this situation

**Review by mvo5 — APPROVED** [review]
Thank you!

**Pull request merged**